petri/src/vm/mod.rs

```diff
@@ -663,6 +664,19 @@ impl<T: PetriVmmBackend> PetriVmBuilder<T> {
         self
     }
 
+    /// Run the VM with Enable VMBus relay enabled
+    pub fn with_device_tree_overrides(
+        mut self,
+        device_tree_overrides: DeviceTreeOverrideParams,
+    ) -> Self {
+        self.config
+            .firmware
+            .openhcl_config_mut()
+            .expect("Device tree overrides are only supported for OpenHCL firmware.")
```

Comment:
I was suprised to see this assumption, but I guess it makes sense: we only generate the device tree for OpenHCL VMs.